modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java

```diff
@@ -390,7 +390,9 @@ private ServiceDeploymentEntity storeNewDeployServiceEntity(DeployTask deployTas
         entity.setDeployResources(new ArrayList<>());
         entity.setServiceVendor(deployTask.getServiceVendor());
         entity.setServiceDeploymentState(ServiceDeploymentState.DEPLOYING);
-        entity.setServiceTemplateId(deployTask.getServiceTemplateId());
+        ServiceTemplateEntity serviceTemplateEntity =
+                serviceTemplateStorage.getServiceTemplateById(deployTask.getServiceTemplateId());
```

Comment:
Is it really needed to fetch the full entity here to set it?it should be ideally sufficient to simply create a new ServiceTemplateEntity with only ID value set and then the hibernate will automatically link to the correct template entity